mautic/mautic
# [UX] Clone report rows

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

**Did you already catch yourself adding manually, one by one, similar rows in reports?**

When creating reports for bounces, I commonly use several filters to include a certain type (like "no more storage space")
but the DNC Comment can have these keywords in the case:

- out of quota
- inbox full
- no storage space

you might also want to create custom reports for other reasons
at the end, you keep repeating the same process over and over

**what if you had a "clone row" button? :bulb:**

this feature clones specifically:
- Column 
- Value

the condition is not cloned as you might always want to recheck it

![image](https://github.com/user-attachments/assets/d5989a1c-dc4a-4f8e-9077-1bf3a4a798fc)









---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Reports
3. Create a new report
4. On Data source, select Do Not Contact
5. Go to the Data tab
6. Add a Filter
7. Add the DNC Comment filter
8. Fill the Value "inbox"
9. Click the Clone button
10. See a cloned row
